docs/spec/type_system.rst

```diff
@@ -4,5 +4,471 @@
+
+They work similar to unsigned integers, but the range is different. For a given bitwidth :math:`b`, the minimum is
+:math:`-(2^{b - 1})` and the maximum is :math:`2^{b - 1} - 1`. The following table listsa all range of all supported
+signed integer types.
+
++-----------+----------------------------+---------------------------+
+| Type      | Minimum                    | Maximum                   |
++===========+============================+===========================+
+| ``int8``  | -128                       | 127                       |
++-----------+----------------------------+---------------------------+
+| ``int16`` | -32,768                    | 32,767                    |
++-----------+----------------------------+---------------------------+
+| ``int32`` | -2,147,483,648             | 2,147,483,647             |
++-----------+----------------------------+---------------------------+
+| ``int64`` | -9,223,372,036,854,775,808 | 9,223,372,036,854,775,807 |
++-----------+----------------------------+---------------------------+
+
+Therefore, all values of all supported signed integer types can be represented by ``int64``.
+
+Floats
+~~~~~~
+Floating point values (often just called "floats") are used to store quantitative data where precision is often only
+relatively important. For example, if you want to know the age of the universe in milliseconds (around 4.35e+20), a
+millisecond more or less might not make the difference.
+
+The floating point types ``float16``, ``float32``, and ``float64`` are defined in `IEEE 754`_ and are called
+``binary{16, 32, 64}`` there.
+
++-------------+---------------+---------------+
+| Type        | Fraction Bits | Exponent Bits |
++=============+===============+===============+
+| ``float16`` | 11            | 5             |
++-------------+---------------+---------------+
+| ``float32`` | 24            | 8             |
++-------------+---------------+---------------+
+| ``float64`` | 53            | 11            |
++-------------+---------------+---------------+
+
+Values are stored, depending on the bit-pattern in the exponent bits, as:
+
+- **normalized:** :math:`(-1)^\text{signbit} \times 1.\text{fraction}_2 \times 2^{\text{exponent}_2 - \text{expbias}}`
+- **subnormals:** :math:`(-1)^\text{signbit} \times 0.\text{fraction}_2 \times 2^{1 - \text{expbias}}`
+- **zero:** :math:`(-1)^\text{signbit} \times 0`
+- **infinity:** :math:`(-1)^\text{signbit} \times \infty`
+
+where :math:`\text{expbias} = 2^{\#\text{exponentbits}}-1`.
+
+.. important::
+
+   `Signaling NaN`_ values are discouraged and should not be used!
+
+.. important::
+
+   `NaN payloads`_ are not handled and should not be used. The `IEEE 754`_ declares them as optinal and hardware and
+   software may wipe them anyway, so portable code cannot make use this data.
+
+For each of these categories, we can represent all values of ``float{16, 32}`` by using ``float64``. So we normalize all
+floating point types to ``float64``.
+
+Decimal
```

Comment:
Do we offer any decimal support?

Reply:
good point. I've double-checked the semantic and updated the section. Would advice against using them.